contrib/upgrade.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+set -e
+
+# This is an update script for gitea deployed from the binary distribution
+# from dl.gitea.io on linux as systemd service. It performs backup and updates
+# Gitea in place.
+# Depends on: bash, curl, xz, sha256sum, gpg, which. optionally jq.
+# Usage:      [environment vars] upgrade.sh [version]
+#   See section below for available environment vars.
+#   When no version is specified, updates to the latest release.
+# Examples:
+#   upgrade.sh 1.15.10
+#   giteahome=/opt/gitea giteaconf=$giteahome/app.ini upgrade.sh
+
+# apply variables from environment
+: ${giteabin:=/usr/local/bin/gitea}
+: ${giteahome:=/var/lib/gitea}
+: ${giteaconf:=/etc/gitea/app.ini}
+: ${giteauser:=git}
+: ${sudocmd:=sudo}
+: ${arch:=linux-amd64}
```

Comment:
Where is the `:` coming from?

Reply:
this is a bash builtin noop. if omitted, bash will try to execute the assigned value :grimacing: `giteabin=${giteabin:-/usr/local/bin/gitea}` is probably more readable.